packages/opentelemetry-api/src/metrics/Meter.ts

```diff
@@ -41,9 +50,27 @@ export interface Meter {
   createCounter(name: string, options?: MetricOptions): Counter;
 
   /**
-   * Creates a new `Observer` metric.
+   * Creates a new `ValueObserver` metric.
    * @param name the name of the metric.
    * @param [options] the metric options.
+   * @param [callback] the observer callback
    */
-  createObserver(name: string, options?: MetricOptions): Observer;
+  createValueObserver(
+    name: string,
+    options?: MetricOptions,
+    callback?: (observerResult: ObserverResult) => void
+  ): ValueObserver;
+
+  /**
+   * Creates a new `BatchObserver` metric, can be used to update many metrics
+   * at the same time and when operations needs to be async
+   * @param name the name of the metric.
+   * @param callback the batch observer callback
+   * @param [options] the metric batch options.
+   */
+  createBatchObserver(
```

Comment:
Why are the argument orders different? This is confusing

Reply:
Because some of them are optional and some not. In batch you have to create name and callback, options in most cases you will not need to define. In observer only name is required, then options and still callback might not be needed. Also the options for ValueObserver are not there same, whereas ValueObserver and UpDownCounter has the same options and the same order of params.

Reply:
Could overload templates be added to this function to handle the different optional combos?